api/konnect/v1alpha1/konnect_extension_types.go

```diff
@@ -180,20 +181,50 @@ type KonnectExtensionStatus struct {
 	// +kubebuilder:validation:Optional
 	DataPlaneClientAuth *DataPlaneClientAuthStatus `json:"dataPlaneClientAuth,omitempty"`
 
+	// Konnect contains the status information related to the Konnect Control Plane.
+	//
+	// +kubebuilder:validation:Optional
+	Konnect *KonnectExtensionControlPlaneStatus `json:"konnect,omitempty"`
+
 	// Conditions describe the current conditions of the KonnectExtensionStatus.
 	// Known condition types are:
 	//
-	// * "Accepted"
-	// * "SecretProvisioned"
-	// * "ControlPlaneRefValid"
-	//
 	// +optional
 	// +listType=map
 	// +listMapKey=type
 	// +kubebuilder:validation:MaxItems=8
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
+// ClusterType is the type of the Konnect Control Plane.
+type ClusterType string
```

Comment:
```suggestion
// KonnectExtensionClusterType is the type of the Konnect Control Plane.
type KonnectExtensionClusterType string
```

Ditto about sharing the package scope with other types.